pkg/server/commands/batch_check_command_test.go

```diff
@@ -0,0 +1,220 @@
+
+	"github.com/openfga/openfga/internal/cachecontroller"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/internal/graph"
+	mockstorage "github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestBatchCheckCommand(t *testing.T) {
+	maxChecks := uint32(50)
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+	ds := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	mockCheckResolver := graph.NewMockCheckResolver(mockController)
+	model := testutils.MustTransformDSLToProtoWithID(`
+		model
+			schema 1.1
+		type user
+		type doc
+			relations
+				define viewer: [user]
+	`)
+	ts, err := typesystem.NewAndValidate(context.Background(), model)
+	require.NoError(t, err)
+
+	cmd := NewBatchCheckCommand(
+		ds,
+		mockCheckResolver,
+		ts,
+		WithBatchCheckMaxChecksPerBatch(maxChecks),
+		WithBatchCheckCommandCacheController(cachecontroller.NewNoopCacheController()),
+	)
+
+	t.Run("calls_check_once_for_each_tuple_in_batch", func(t *testing.T) {
+		numChecks := 50
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		for i := 0; i < numChecks; i++ {
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: fmt.Sprintf("fakeid%d", i),
+			}
+		}
+
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).
+			Times(numChecks).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, _, err := cmd.Execute(context.Background(), params)
+		require.NoError(t, err)
+		require.Equal(t, len(result), numChecks)
+	})
+
+	t.Run("returns_a_result_for_each_correlation_id", func(t *testing.T) {
+		numChecks := 10
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		var ids []string
+		for i := 0; i < numChecks; i++ {
+			correlationID := fmt.Sprintf("fakeid%d", i)
+			ids = append(ids, correlationID)
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: correlationID,
+			}
+		}
+
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).
+			Times(numChecks).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, _, err := cmd.Execute(context.Background(), params)
+		require.NoError(t, err)
+
+		// Make sure all correlation ids are present in the response
+		for _, id := range ids {
+			_, ok := result[CorrelationID(id)]
+			require.True(t, ok)
+		}
```

Comment:
Really minor but might be worth it to compare the quantity of correlation IDs to the quantity of response keys. This will ensure that no additional keys exist in the response.